eng/pipelines/templates/steps/analyze.yml

```diff
@@ -43,6 +43,14 @@ steps:
     failOnStderr: true
     workingDirectory: 'sdk/${{parameters.ServiceDirectory}}'
 
+  - script: |
+      echo scanning copyright headers in $(pwd)
+      grep -L -r --include LICENSE -P "Copyright (\d{4}|\(c\)) Microsoft" . | tee >&2
+    displayName: 'License Check'
+    condition: succeededOrFailed()
+    failOnStderr: true
```

Comment:
Will grep set the correct exit code?

Reply:
I believe so, I used the same command as the `Copyright Header Check` step above and have seen this one fail as expected.

Reply:
Adding `./` breaks the grep command so it doesn't work, because the `--include` value is supposed to be a glob, which it then searches recursively for in `.`.

```
# Example with LICENSE.txt in net repo
⇉ ⇉ ⇉ grep -L -r --include LICENSE.txt -P "donotmatch" .  | tee >&2
./samples/CloudClipboard/CloudClipboard/wwwroot/lib/jquery-validation-unobtrusive/LICENSE.txt
./samples/CloudClipboard/CloudClipboard/wwwroot/lib/jquery/LICENSE.txt
./samples/AppSecretsConfig/wwwroot/lib/jquery-validation-unobtrusive/LICENSE.txt
./samples/AppSecretsConfig/wwwroot/lib/jquery/LICENSE.txt
./LICENSE.txt

# Should succeed but fails
⇉ ⇉ ⇉ grep -L -r --include ./LICENSE.txt -P "Copyright (\d{4}|\(c\)) Microsoft" .
⇉ ⇉ ⇉ echo $?
1
```

I think the reason this may be succeeding is because `pipefail` is not set, so piping the command to `tee` masks the exit code.